org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/model/JDIStackFrame.java

```diff
@@ -1730,4 +1732,12 @@ public void forceReturn(IJavaValue value) throws DebugException {
 	public void setIsTop(boolean isTop) {
 		this.fIsTop = isTop;
 	}
+
+	@Override
+	public Category getCategory() {
+		if (fCategory == null) {
+			fCategory = JDIDebugPlugin.getDefault().getStackFrameCategorizer().categorize(this);
```

Comment:
that's a better cache mechanism :-) as i guess this code is called from different threads it needs some ynchronization

Reply:
synchronization added